v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/transforms/FileFormatFactory.java

```diff
@@ -0,0 +1,137 @@
+ * format.
+ */
+@AutoValue
+public abstract class FileFormatFactory
+    extends PTransform<PCollection<KV<String, String>>, POutput> {
+
+  /** Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(FileFormatFactory.class);
+
+  /** Valid file formats. */
+  public static final String EXPECTED_FILE_FORMAT =
+      Joiner.on(',').join(Arrays.asList(WriteToGCSUtility.FileFormat.values()));
+
+  public static WriteToGCSBuilder newBuilder() {
+    return new AutoValue_FileFormatFactory.Builder();
+  }
+
+  public abstract KafkaToGCSOptions options();
+
+  public abstract String outputFileFormat();
+
+  @Override
+  public POutput expand(PCollection<KV<String, String>> records) {
+    POutput output = null;
+
+    String errorMessage =
+        "Invalid output format:"
+            + outputFileFormat()
+            + ". Supported output formats:"
+            + FileFormatFactory.EXPECTED_FILE_FORMAT;
+
+    /*
+     * Call the function to check file format passed by user is valid.
+     */
+    if (!WriteToGCSUtility.isValidFileFormat(outputFileFormat())) {
+      LOG.info(errorMessage);
+      throw new IllegalArgumentException(errorMessage);
+    }
+
+    WriteToGCSUtility.FileFormat outputFileFormat =
+        WriteToGCSUtility.FileFormat.valueOf(outputFileFormat());
+
+    /*
+     * Calls appropriate class Builder to performs PTransform based on user provided File Format.
+     */
+    switch (outputFileFormat) {
+      case TEXT:
+        output =
+            records.apply(
+                "Write Text File(s)",
+                WriteToGCSText.newBuilder()
+                    .withOutputDirectory(options().getOutputDirectory())
+                    .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                    .withNumShards(options().getNumShards())
+                    .withTempLocation(options().getTempLocation())
+                    .build());
+        break;
+
+      case AVRO:
+        output =
+            records.apply(
+                "Write Avro File(s)",
+                WriteToGCSAvro.newBuilder()
+                    .withOutputDirectory(options().getOutputDirectory())
+                    .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                    .withNumShards(options().getNumShards())
+                    .withTempLocation(options().getTempLocation())
+                    .build());
+        break;
+
+      case PARQUET:
+        output =
+            records.apply(
+                "Write Parquet File(s)",
+                WriteToGCSParquet.newBuilder()
+                    .withOutputDirectory(options().getOutputDirectory())
+                    .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                    .withNumShards(options().getNumShards())
+                    .build());
+        break;
+
+      default:
+        LOG.info(errorMessage);
```

Comment:
What if a new file format is added to FileFormat but no implementation is provided. In that case, the errorMessage will be confusing. For example, if ORC is supported by adding in FileFormat but no implementation is available in the switch, the error message would be something like:

Invalid output format: ORC. Supported output formats: AVRO, TEXT, PARQUET, ORC